### ISSUE TITLE
We have identified a freezing issue on iOS

### ISSUE THREAD

**Issue by MolierQueen** [firebase/firebase-ios-sdk#12799]
### Description

![image](https://github.com/firebase/firebase-ios-sdk/assets/11712198/92147498-7f1f-4f41-aaf3-848ca12118ac)
![image](https://github.com/firebase/firebase-ios-sdk/assets/11712198/3cd6b329-c040-45ef-b78a-836b4cbc3b24)

Thread 20
![image](https://github.com/firebase/firebase-ios-sdk/assets/11712198/8509fc55-0b08-4de5-82ce-2748938088f0)

Thread 64
![image](https://github.com/firebase/firebase-ios-sdk/assets/11712198/238f7514-fc20-432f-ad03-da811097648f)

### Firebase SDK Version

10.17

### Xcode Version

15.1

### Installation Method

CocoaPods

### Firebase Product(s)

Analytics

### Targeted Platforms

iOS

### If using Swift Package Manager, the project's Package.resolved

<details>
<summary>Expand <code>Package.resolved</code> snippet</summary>
<br>

```json

Replace this line with the contents of your Package.resolved.

```

</details>

### If using CocoaPods, the project's Podfile.lock

<details>
<summary>Expand <code>Podfile.lock</code> snippet</summary>
<br>

```yml

Replace this line with the contents of your Podfile.lock!

```

</details>

**Comment by rizafran** [comment]
Thanks for filing, @MolierQueen. Per checking, your stack trace looks related to #12129 and is already fixed on v10.20.0. With this, you may try updating your SDK to 10.20.0.

**Comment by MolierQueen** [comment]
Thank you very much for your reply, we will upgrade the version and try again
